### ISSUE TITLE
mod not working correctly on some numbers

### ISSUE THREAD

**Issue by wanderer** [indutny/bn.js#45]
```javascript
var a = new BN('945304eb96065b2a98b57a48a06ae28d285a71b5', 'hex');
var b = new BN('fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe', 'hex');

a.mul(b).mod(a) ///gives 945304eb96065b2a98b57a48a06ae28d285a71b5 should be '0'
```

mod should be zero in this case but it is not

**Comment by indutny** [comment]
Looking...

**Comment by indutny** [comment]
Confirmed, bug.

**Comment by indutny** [comment]
Fixed in 7320346, thank you!
